src/main/java/spoon/support/visitor/java/JavaReflectionVisitorImpl.java

```diff
@@ -131,46 +187,70 @@ public void visitPackage(Package aPackage) {
 		if (clazz.getPackage() != null) {
 			clazz.getPackage();
 		}
-		for (Type anInterface : clazz.getGenericInterfaces()) {
-			visitTypeReference(CtRole.INTERFACE, anInterface);
-		}
-		for (Annotation annotation : clazz.getDeclaredAnnotations()) {
-			visitAnnotation(annotation);
+		try {
+			for (Type anInterface : clazz.getGenericInterfaces()) {
+				visitTypeReference(CtRole.INTERFACE, anInterface);
+			}
+		}  catch (Throwable ignore) {
+			// partial classpath
 		}
-		for (Constructor<?> constructor : clazz.getDeclaredConstructors()) {
-			if (Modifier.isPrivate(constructor.getModifiers())) {
-				Class<?>[] paramTypes = constructor.getParameterTypes();
-				if (paramTypes.length == 2 && paramTypes[0] == String.class && paramTypes[1] == int.class) {
-					//ignore implicit enum constructor
+		try {
+			for (Annotation annotation : clazz.getDeclaredAnnotations()) {
+				visitAnnotation(annotation);
+			}
```

Comment:
This for cycle seems to be copy past here by mistake. It is already above. Please fix it

Reply:
I don't understand it is the only loop on clazz.getDeclaredAnnotations() for visitEnum(Class<T> clazz).
Did I miss something?

Reply:
See line 142-144 in this file

Reply:
It is not the same method

Reply:
I see. First one processes Interface and second one processes Enum. You are right. It is correct. Thanks for explanation.